google-cloud-spanner/lib/google/cloud/spanner/service.rb

```diff
@@ -376,12 +389,10 @@ def inspect
 
         protected
 
-        def default_headers
-          { "google-cloud-resource-prefix" => "projects/#{@project}" }
-        end
-
-        def default_options
-          Google::Gax::CallOptions.new kwargs: default_headers
+        def default_options_from_session session_name
+          default_prefix = session_name.split("/sessions/").first
```

Comment:
For Data API calls it will always be the database name. In #1237 we are instructed that those are the only calls that require the header.